### DISCUSSION TITLE
How to exchange the LWA authorization code for a LWA refresh token

### DISCUSSION THREAD

**Discussion by MMV-Code** [abuzuhri/Amazon-SP-API-CSharp#193] [Q&A]
Hi everyone.

I have been implementing the  Website Authorization Workflow for my application, and in the [step 4 ](https://developer-docs.amazon.com/sp-api/docs/website-authorization-workflow#step-4-your-application-exchanges-the-lwa-authorization-code-for-a-lwa-refresh-token) I need to exchange the LWA authorization code for a LWA refresh token.

I have seen a method that I thought was the one that I should use, which is in the class AuthorizationService the GetAccessTokenFromCodeAsync, but this method is returning to me the refresh token as null, checking the implementation, it is sending the grant type as client credentials, and the scopes for migrations, etc, etc, so, it makes me think that is not the one that I should use.

I would like to ask you if there is some method to do this process implemented in the sdk or should I do this out of the sdk scope.

Thank you!

Edit:
I'm using TokenGeneration class on method GetAccessTokenFromCodeAsync for now and it works, but just in case this is not correct (due to future changes or something like that) please let me know.

**Comment by abuzuhri** [comment]
Hi @MMV-Code [MMV-Code](https://github.com/MMV-Code)


Can you share code you use

**Reply by tank104** [reply]
@abuzuhri this is fixed with the change I added which allows you to get the Refresh token (GetRefreshTokenFromCodeAsync)

**Reply by abuzuhri** [reply]
The code required in the parameter from where you get it ?

**Reply by tank104** [reply]
Not sure I understand - but it was this PR: https://github.com/abuzuhri/Amazon-SP-API-CSharp/pull/255
